### ISSUE TITLE
Type specifiers with generic args in certain positions should generate errors that reference the turbofish

### ISSUE THREAD

**Issue by aathan** [rust-lang/rust#114112]
### Code

```Rust
I'm having trouble creating a concise example, but perhaps the error below is enough. If the error at least mentioned turbofish it'd have saved a bunch of heartache for a noob.
```

### Current output

```Shell
error: expected one of `)`, `,`, `@`, or `|`, found `<`
   --> ....
    |
143 |             Ok(SomeEnum<STRICT>::SomeVariant { result, .. }) => { ... },
    |                           ^
    |                           |
    |                           expected one of `)`, `,`, `@`, or `|`
    |                           help: missing `,`
```

### Desired output

```Shell
" if SomeEnum<STRICT> is a type reference, use SomeEnum::<STRICT> instead"
```

### Rationale and extra context

Rust is elsewhere good about suggesting the turbofish.

**Comment by MU001999** [comment]
An example ([play](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=06e73dc7feb10a67f2e4c0f2cbbb8b18)):
```rust
enum E<T> {
    A(T)
}

fn main() {
    match E::A(1) {
        E<i32>::A(v) => {
            println!("{v:?}");
        },
    }
}
```

**Comment by estebank** [comment]
To summarize, the parser needs to keep into account a lack of turbofish in patterns. It already does in expressions, and the same strategy can likely be used.
